components/Modal/UserAccount.js

```diff
@@ -182,7 +182,17 @@ class UserAccount extends React.Component {
   }
 
   encryptPassword(password) {
-    return cockpit.spawn(["openssl", "passwd", "-6", "-stdin"], { err: "message" }).input(password);
+    // inspired by https://github.com/cockpit-project/cockpit/blob/master/pkg/lib/python.js
+    const pyinvoke = [
+      "sh",
+      "-ec",
+      "exec $(which /usr/libexec/platform-python 2>/dev/null || which python3 2>/dev/null || which python) $@",
+      "--",
+      "-",
+      "-c"
+    ];
```

Comment:
This is unsafe now -- the password can legitimately contain `"`, or newlines and such. This wasn't a problem with feeding the password through stdin. I think it's probably better to keep that behaviour, and don't feed the Python source code through stdin, but on the CLI instead. This is easier than with python.js, as you have complete control over the entire command line. Can you please try with

```js
return cockipt.script(
    "$(which /usr/libexec/platform-python 2>/dev/null || which python3 2>/dev/null || which python) -c 'import sys, crypt; print(crypt.crypt(sys.stdin.readline().strip(), crypt.mksalt(crypt.METHOD_SHA512)))' ",
    { err: "message" }).input(password);
```

Reply:
Thank you so much @martinpitt! :+1: :+1:  Updating now.